### DISCUSSION TITLE
_transform_output_shape: useless .clone() that increases memory consumption?

### DISCUSSION THREAD

**Discussion by Hendrik-code** [kornia/kornia#3487] [Potential Issue]
in kornia.augmentation.utils.helper.py:
```
def _transform_output_shape(
    output: Tensor, shape: Tuple[int, ...], *, reference_shape: Optional[Tensor] = None
) -> Tensor:
    r"""Collapse the broadcasted batch dimensions an input tensor to be the specified shape.

    Args:
        output: Tensor
        shape: List/tuple of int
        reference_shape: Tensor representation of shape to control which dimensions are collapsed.

    Returns:
        Tensor

    """
    out_tensor = output.clone()

    for dim in range(len(out_tensor.shape) - len(shape)):
        idx = 0
        if reference_shape is not None and out_tensor.shape[0] == reference_shape[0] != 1 and len(shape) > 2:
            idx = 1
        if out_tensor.shape[idx] != 1:
            raise AssertionError(f"Dimension {dim} of input is expected to be 1, got {out_tensor.shape[idx]}")
        out_tensor = out_tensor.squeeze(idx)

    return out_tensor
```

The first line of this function clones the entire tensor.
My question: is this necessary?

I am dealing with lots of custom written 3D augmentations on large images, and this leads to crazy high GPU memory consumption.
If I replace this line
`out_tensor = output.clone()`
with
`out_tensor = output.detach()`

everything runs fine, and the memory consumption is much lower.
Tested on:
- kornia 0.8.2
- Ubuntu 20.04

Pending tests:
- Using only kornia-unmodified augmentations

Here is a test of mine with the current code:
<img width="657" height="944" alt="image" src="https://github.com/user-attachments/assets/d5ee5c59-5897-42a5-a44d-96c45a58e6a0" />

With the modified code:
<img width="645" height="1024" alt="image" src="https://github.com/user-attachments/assets/99425751-3dc4-4d97-8411-7e9a1ff605a4" />

**Comment by edgarriba** [comment]
are you open to submit a PR with the fix ? also just to make sure that test pass and it doesn't break anything else

**Reply by Hendrik-code** [reply]
sure thing, will do that :)

**Reply by Hendrik-code** [reply]
@edgarriba first time contributing here. You want me to open a full issue first, get assigned, then create the PR or just do the PR directly from this?

**Comment by edgarriba** [comment]
https://github.com/kornia/kornia/issues/3494

**Comment by RizgarOzan** [comment]
For anyone landing here later: this was fixed in #3510 (closing #3494). `_transform_output_shape` now starts from `out_tensor = output` with no copy, and the change is in v0.8.3 and later (`kornia/augmentation/utils/helpers.py`).

One note on the workaround in the question: `output.detach()` also removes the copy, but it cuts the autograd graph, so gradients stop flowing back through the augmentation. The version that was merged just skips the copy, so gradients still work. On kornia 0.8.2 it is better to patch that line to `out_tensor = output` than to use `detach()`.
